### DISCUSSION TITLE
V6 register validation not working

### DISCUSSION THREAD

**Discussion by bhuvaneshdesai** [react-hook-form/react-hook-form#2266] [Q&A]
i am trying to migrate to v6 of react-hook-form
using material UI textField component passed as render to Controller component
`
<TextField
                name={name}
                placeholder={placeholder}
                disabled={disabled}
                // error={!!errors[name]}
                inputRef={register(validate)}
            />
{errors[name] ? (
                    <span className="error-feedback font-wrap">
                        {errors[name].message}
                    </span>
                ) : null}
`
here above validate is ` {required: "Required"}`
i don't want to register using useEffect
earlier this was working totally fine

https://codesandbox.io/s/romantic-bardeen-pn31y?file=/src/App.js

**Comment by bluebill1049** [comment]
provide a codesanbox

**Reply by bhuvaneshdesai** [reply]
code sandbox link added

**Reply by bhuvaneshdesai** [reply]
@bluebill1049 ☝️

**Comment by EvgenBabenko** [comment]
@bhuvaneshdesai 
i think my example will work how you expected, without useEffect
https://codesandbox.io/s/fervent-archimedes-4176j?file=/src/App.jsx

**Reply by EvgenBabenko** [reply]
and in TextFields just disable prop required, so you can see only yup validation

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/ancient-water-fcfxk

**Reply by bhuvaneshdesai** [reply]
Thankyou for reply
but my code is way more complex than a simple form(like everything is dynamic and all).
that is the reason i am using ref way to validate and trying to avoid validation schema or useEffect
Plus it was working in v5 not sure if its a bug or something in v6
